### PR TITLE
Add oVirt platform status

### PR DESCRIPTION
**Pull request by rgolangh**
Add oVirt platform status

enhancement: https://github.com/openshift/enhancements/pull/61

### CODE REVIEW

**Comment by deads2k** [comment]
Please link the enhancement to this PR so we can understand the impact this has on 4.3

**Comment by deads2k** [comment]
> Please link the enhancement to this PR so we can understand the impact this has on 4.3

Can you link the enhancement please

**Comment by sdodson** [comment]
/approve
Seems inline with other platform definitions.

@abhinavdahiya PTAL

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -199,6 +203,26 @@ type OpenStackPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+// OvirtPlatformStatus holds the current status of the  oVirt infrastructure provider.
+type OvirtPlatformStatus struct {
+	// apiServerInternalIP is an IP address to contact the Kubernetes API server that can be used
+	// by components inside the cluster, like kubelets using the infrastructure rather
+	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
+	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	APIServerInternalIP string `json:"apiServerInternalIP,omitempty"`
```
Comment (abhinavdahiya) [inline config/v1/types_infrastructure.go]: so these fields are required but `omitempty` ??

@sttts / @deads2k any guidance here?

Comment (soltysh) [reply config/v1/types_infrastructure.go]: `omitempty` with string will cause they are not printed when empty, but it doesn't make any difference during parsing. I don't think we have clear guidance for it, though.

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -130,6 +130,10 @@ type PlatformStatus struct {
 	// OpenStack contains settings specific to the OpenStack infrastructure provider.
 	// +optional
 	OpenStack *OpenStackPlatformStatus `json:"openstack,omitempty"`
+
+	// Ovirt contains settings specific to the oVirt infrastructure provider.
+	// +optional
```
Comment (sttts) [inline config/v1/types_infrastructure.go]: @damemi one needs the kubebuilder optional tag as well, doesn't one?

Comment (sttts) [reply config/v1/types_infrastructure.go]: though it does not matter here as the default is optional already

File: config/v1/types_infrastructure.go (hunk 1 above)
Comment (sttts) [reply config/v1/types_infrastructure.go]: omitempty for optional fields is often a good idea. Usually only if you want to distinguish undefined and zero value, you don't use omitempty.

Comment (abhinavdahiya) [reply config/v1/types_infrastructure.go]: > https://github.com/openshift/api/blob/master/config/v1/doc.go#L5

all fields are kindof optional at package level, and this is similar to other structs, so following that seems fine to me.

**Pull request merged**